pkg/server/handler.go

```diff
@@ -43,18 +43,40 @@ func NewMsgHandler(logger logging.Logger, amqp network.IAmqpReceiver, thingContr
 
 // Start starts to listen messages
 func (mc *MsgHandler) Start(started chan bool) {
+	err := mc.start(started, make(chan network.InMsg))
+	if err != nil {
+		mc.logger.Error(err)
+	}
+}
+
+func (mc *MsgHandler) start(started chan bool, msgChan chan network.InMsg) error {
 	mc.logger.Debug("message handler started")
-	msgChan := make(chan network.InMsg)
+	if started == nil {
+		return errors.New("missing channel")
```

Comment:
What do you think about adding this error at the top of the file? See this [example](https://github.com/docker/distribution/blob/master/reference/reference.go#L40) :grimacing: